client/servers/manager.go

```diff
@@ -0,0 +1,529 @@
+// Package servers provides an interface for choosing Servers to communicate
```

Comment:
This appears to be the old rpcproxy package code from pre #1735? I know serverlist from #1735 didn't have the desired behavior, but if I remember correctly this code did not pass Go's race detector.

It's also just not very idiomatic code with warnings about keeping serverList on the stack (not sure why when it contains pointers), an atomic used as a "barrier" which I think should just be a lock, and a lock around an atomic.Value.

I can try to review this code and point out individual issues, but it appears I already tried to fix them in #1718 and must have failed? At any rate I don't see the changes from #1718 here, so I'm not sure this code is worth reviewing as its known to have bugs.